src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.ToDoCommand;
+
+/**
+ * This class is responsible for parsing the input provided by the user into the respective commands they represent
+ */
+public class Parser {
+
+    /**
+     * Produces a Command object corresponding to the input provided. Throws an exception if invalid input is provided.
+     * @param input raw string form of a command
+     * @return Command object representing the command in the input
```

Comment:
maybe you can add punctuation behind each parameter description? (as per java coding standards, and for the other javadoc comments as well!) :)